botorch/optim/utils/numpy_utils.py

```diff
@@ -125,7 +124,12 @@ def set_tensors_from_ndarray_1d(
             try:
                 size = tnsr.numel()
                 vals = array[index : index + size] if tnsr.ndim else array[index]
-                tnsr.copy_(as_tensor(vals).to(tnsr).view(tnsr.shape).to(tnsr))
+                tnsr.copy_(
+                    torch.as_tensor(vals, device=tnsr.device, dtype=tnsr.dtype)
+                    .to(tnsr)
+                    .view(tnsr.shape)
+                    .to(tnsr)
```

Comment:
It seems that these `.to()` are noops in this setup given that the tensor is initially constructed on the right device with the right dtype? 
```suggestion
                    torch.as_tensor(vals, device=tnsr.device, dtype=tnsr.dtype)
                    .view(tnsr.shape)
```

Reply:
Actually, let's do the reshaping first and then use `from_numpy`: `torch.from_numpy(vals.reshape(tnsr.shape)).to(tnsr)`. I did some lightweight benchmarking and this generally is a tad faster.

Reply:
The benchmarking explains the 22 minute gap between the two comments :D. I updated the diff directly rather than waiting for back & forth.

Reply:
Looks like this doesn't work if vals is a scalar. Reverting to the previous solution

Reply:
Since there have been change after this comment, is there anything I can still do to assist? In any case, thanks for already implementing the necessary changes :)

Reply:
I think the PR is ready as is. I'll land it after the internal checks pass. Thanks for the contribution!

Reply:
Glad that I could help :)